### ISSUE TITLE
r.table().map().coerceTo(object) error

### ISSUE THREAD

**Issue by wojons** [rethinkdb/rethinkdb#2802]
Doing something like the above but will show it again here.

```js
r.table().map().coerceTo('object')
```

```
RqlRuntimeError: Cannot coerce STREAM to OBJECT in:
```

but if i do it like

```js
r.table().map().coerceTo('array').coerceTo('object')
```

That works fine i feel like if the system comes across this type of error it where changing it to array and then object would solve it it should do it with out having throw out an error. I guess this should be considered one of the first query optimizes

**Comment by mlucy** [comment]
I think we should do this as described.

@danielmewes -- how would you feel about bumping this to polish?

**Comment by mlucy** [comment]
Er, nevermind, beat me to it ^.^

**Comment by danielmewes** [comment]
I have no objections. We should overload `object` for streams.

**Comment by danielmewes** [comment]
Marking settled as suggested.

**Comment by encryptio** [comment]
Fixed in 2fcc0b5 in `next`, CL 3211
